recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,157 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class SociConan(ConanFile):
+    name = "soci"
+    homepage = "https://github.com/SOCI/soci"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "The C++ Database Access Library "
+    topics = ("C++", "database-library", "oracle", "postgresql", "mysql", "odbc", "db2", "firebird", "sqlite3", "boost" )
+    license = "Boost"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    options = {
+        "fPIC":       [True, False],
+        "static":     [True, False],
+        "cxx11":      [True, False],
+        "empty":      [True, False],
+        "shared":     [True, False],
+        "tests":      [True, False],
+        "sqlite3":    [True, False],
+        "db2":        [True, False],
+        "odbc":       [True, False],
+        "oracle":     [True, False],
+        "firebird":   [True, False],
+        "mysql":      [True, False],
+        "postgresql": [True, False],
+        "boost":      [True, False],
+        "valgrind":   [True, False]
+    }
+
+    default_options = {
+        "fPIC":       True,
+        "static":     True,
+        "cxx11":      True,
+        "empty":      True,
+        "shared":     False,
+        "tests":      False,
+        "sqlite3":    False,
+        "db2":        False,
+        "odbc":       False,
+        "oracle":     False,
+        "firebird":   False,
+        "mysql":      False,
+        "postgresql": False,
+        "boost":      False,
+        "valgrind":   False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        prefix  = "Dependencies for "
+        message = " not configured in this conan package, some features will be disabled."
+
+        if self.options.sqlite3:
+            self.requires("sqlite3/3.33.0")
+        # ToDo add the missing dependencies for the backends
+        if self.options.db2:
+            self.output.warn(prefix + "DB2" + message)
+        if self.options.odbc:
+            self.output.warn(prefix + "ODBC" + message)
+        if self.options.oracle:
+            self.output.warn(prefix + "ORACLE" + message)
+        if self.options.firebird:
+            self.output.warn(prefix + "FIREBIRD" + message)
+        if self.options.mysql:
+            self.output.warn(prefix + "MYSQL" + message)
+        if self.options.postgresql:
+            self.output.warn(prefix + "POSTGRESQL" + message)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
```

Comment:
for consistency, we like to have this after the configure cmake, but do not feel obliged